constraints.txt

```diff
@@ -7,6 +7,10 @@ scipy<1.11; python_version<'3.12'
 # See https://github.com/Qiskit/qiskit/issues/12655 for current details.
 scipy==1.13.1; python_version=='3.12'
 
+# Rustworkx 0.15.0 contains a bug that breaks graphviz-related tests.
```

Comment:
Is this then unrelated to the NumPy deprecation warning?

Reply:
Yes, it's just that Rustworkx 0.14 wasn't compatible with Numpy 2 and Rustworkx 0.15 is, so the release caused us to start pulling in Numpy 2 in CI.  We didn't see the failures before because not only do you need Numpy 2, you also need to have the test suite generate a warning out of `np.linalg._linalg`, which we don't do on most platforms, but there are some about numerical-precision things on macOS ARM specifically, and we weren't running the Numpy 2 tests on macOS ARM previously because it wasn't available to us at the time.